product_manufacturer/models/product_template.py

```diff
@@ -0,0 +1,95 @@
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    manufacturer_id = fields.Many2one(
+        comodel_name="res.partner",
+        compute="_compute_manufacturer_info",
+        inverse="_inverse_manufacturer_info",
+        store=True,
+    )
+    manufacturer_pname = fields.Char(
+        compute="_compute_manufacturer_info",
+        inverse="_inverse_manufacturer_info",
+        store=True,
+        string="Manufacturer Product Name",
+    )
+    manufacturer_pref = fields.Char(
+        compute="_compute_manufacturer_info",
+        inverse="_inverse_manufacturer_info",
+        store=True,
+        string="Manufacturer Product Code",
+    )
+    manufacturer_purl = fields.Char(
+        compute="_compute_manufacturer_info",
+        inverse="_inverse_manufacturer_info",
+        store=True,
+        string="Manufacturer Product URL",
+    )
+
+    @api.depends(
+        "product_variant_ids",
+        "product_variant_ids.manufacturer_id",
+        "product_variant_ids.manufacturer_pname",
+        "product_variant_ids.manufacturer_pref",
+        "product_variant_ids.manufacturer_purl",
+    )
+    def _compute_manufacturer_info(self):
+        unique_variants = self.filtered(
+            lambda template: len(template.product_variant_ids) == 1
+        )
+        for template in unique_variants:
+            template.manufacturer_id = template.product_variant_ids.manufacturer_id
+            template.manufacturer_pname = (
+                template.product_variant_ids.manufacturer_pname
+            )
+            template.manufacturer_pref = template.product_variant_ids.manufacturer_pref
+            template.manufacturer_purl = template.product_variant_ids.manufacturer_purl
+        for template in self - unique_variants:
+            template.manufacturer_id = False
+            template.manufacturer_pname = False
+            template.manufacturer_pref = False
+            template.manufacturer_purl = False
+
+    def _inverse_manufacturer_info(self):
+        for template in self:
+            if len(template.product_variant_ids) == 1:
+                template.product_variant_ids.manufacturer_id = template.manufacturer_id
+                template.product_variant_ids.manufacturer_pname = (
+                    template.manufacturer_pname
+                )
+                template.product_variant_ids.manufacturer_pref = (
+                    template.manufacturer_pref
+                )
+                template.product_variant_ids.manufacturer_purl = (
+                    template.manufacturer_purl
+                )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        """Overwrite creation for rewriting manufacturer information (if set and having
+        only one variant), after the variant creation, that is performed in super.
+        TODO : when migrating in version 16.0, remove the overload of the create function
```

Comment:
This part has been forgotten.
@FrancoMaxime  could you take a look ? 

thanks !

Reply:
@legalsylvain I will make the fix next week if it is ok for you?

Reply:
@legalsylvain I did it, a bit late sorry. Can you take a look at this pr? https://github.com/OCA/product-attribute/pull/1424